src/kiara_plugin/topic_modelling/modules/corpus_preparation.py

```diff
@@ -30,83 +25,221 @@ def create_inputs_schema(self):
+                "doc": "Column name of the column that contains the date. Values in this column need to comply with date format: https://docs.rs/chrono/latest/chrono/format/strftime/index.html.",
+            },
+            "title_col_name": {
+                "type": "string",
+                "doc": "Column name of the values containing publication names or ref/id. This column will be used in the output.",
+            },
+            "corpus_table": {
+                "type": "table",
+                "doc": "The corpus table for which the distribution over time is needed.",
+            },
+        }
+
+    def create_outputs_schema(self):
+        return {"dist_table": {"type": "table", "doc": "The aggregated data table."}}
+
+    def process(self, inputs, outputs) -> None:
+        
+        import duckdb
+        import polars as pl
+        import pyarrow as pa
+
+        agg = inputs.get_value_obj("periodicity").data
+        title_col = inputs.get_value_obj("title_col_name").data
+        time_col = inputs.get_value_obj("date_col_name").data
+        table_obj = inputs.get_value_obj("corpus_table")
+
+        if not agg:
+            raise KiaraProcessingException("No periodicity provided.")
+        
+        if not title_col:
+            raise KiaraProcessingException("No column name provided for titles id/ref.")
+        
+        if not time_col:
+            raise KiaraProcessingException("No column name provided for date.")
+        
+        if not table_obj:
+            raise KiaraProcessingException("No table provided.")
+        
+        if table_obj.is_set:
+
+            sources: Union[pa.Table, None] = table_obj.data
+            
+            assert sources is not None
+
+            sources_col_names = sources.column_names
+
+            if title_col not in sources_col_names:
+
+                raise KiaraProcessingException(
+                    f"Could not find title name/id column '{title_col}' in the table. Please specify a valid column name manually, using one of: {', '.join(sources_col_names)}"
+                )
+            
+            if time_col not in sources_col_names:
+
+                raise KiaraProcessingException(
+                    f"Could not find date column '{time_col}' in the table. Please specify a valid column name manually, using one of: {', '.join(sources_col_names)}"
+                )
+            
+            sources_data: pa.Table = table_obj.data.arrow_table
+            
+            sources_tb: pl.DataFrame = pl.from_arrow(sources_data) # type: ignore
+
+            try:
+                sources_tb = sources_tb.with_columns(
+                    pl.col(time_col).str.strptime(pl.Date, "%Y-%m-%d")
+                )
+
+            except:
+                raise KiaraProcessingException(
+                    f"Could not convert time column to a valid date format. Please check the pattern of source values."
+                )
+
+            if agg == "month":
```

Comment:
I'm going to assume that those sql queries are correct, since them not working would be pretty obvious to you pretty soon. Hard for me to verify without spending a lot of time myself.